src/connectors/impls/ws/server.rs

```diff
@@ -209,19 +213,24 @@ impl Connector for WsServer {
 
                             let (ws_write, ws_read) = ws_stream.split();
 
-                            let ws_writer = WsWriter::new_tls_server(ws_write);
-                            sink_runtime
-                                .register_stream_writer(
-                                    stream_id,
-                                    Some(connection_meta.clone()),
-                                    &ctx,
-                                    ws_writer,
-                                )
-                                .await;
+                            let reader_runtime = if sink_is_connected.load(Ordering::Acquire) {
+                                let ws_writer = WsWriter::new_tls_server(ws_write);
+                                sink_runtime
+                                    .register_stream_writer(
+                                        stream_id,
+                                        Some(connection_meta.clone()),
+                                        &ctx,
+                                        ws_writer,
+                                    )
+                                    .await;
+                                Some(sink_runtime.clone())
+                            } else {
+                                None
+                            };
```

Comment:
I notice this code in quite a few okaces, is there a chance to generalize it? They sem slightly different but somewhat the sasme.

Reply:
it comes up in exactly  places, and the implementations are completely different each time. Don't know if it is worth the generics-juggling.